<a href="https://colab.research.google.com/github/cjhaster/NRSurrogateCatalog/blob/gh-pages/_sources/events/GW190521_030229.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open in Colab"/></a>

In [1]:
%load_ext autoreload
%autoreload 2
%matplotlib inline

import pandas as pd

pd.set_option("display.max_rows", None, "display.max_columns", None)

In [2]:
import pip

try:
    __import__("nrsur_catalog")
except ImportError:
    pip.main(["install", "nrsur_catalog", "-q", ])


# GW190521_030229

Below are some plots for GW190521_030229 from the NRSurrogate Catalog.

In [3]:
from nrsur_catalog import NRsurResult

nrsur_result = NRsurResult.load("GW190521_030229", cache_dir=".nrsur_catalog_cache")
# you can specify a `cache_dir`: folder where data will be downloaded

## Summary

In [4]:
nrsur_result.summary()

Posterior 90% CI  \
Parameter                                                         
$m_1\ [M_{\odot}]$                 ${150.90}_{-12.31}^{+17.37}$   
$m_2\ [M_{\odot}]$                 ${117.28}_{-22.33}^{+14.55}$   
$\mathcal{M}\ [M_{\odot}]$           ${114.27}_{-7.85}^{+8.02}$   
$q$                                    ${0.78}_{-0.21}^{+0.15}$   
$a_1$                                  ${0.66}_{-0.43}^{+0.24}$   
$a_2$                                  ${0.69}_{-0.42}^{+0.23}$   
$\theta_1$                             ${1.58}_{-0.57}^{+0.56}$   
$\theta_2$                             ${1.56}_{-0.61}^{+0.55}$   
$\chi_{\mathrm{eff}}$                  ${0.00}_{-0.18}^{+0.20}$   
$\chi_{\mathrm{p}}$                    ${0.67}_{-0.25}^{+0.19}$   
$\alpha$                               ${3.32}_{-0.12}^{+2.60}$   
$\delta$                               ${0.48}_{-1.38}^{+0.20}$   
$t_c\ [s]$                    ${1242442967.44}_{-0.04}^{+0.00}$   
$d_L\ [Mpc]$                  ${4545.93}_{-1668.08}^{+1594.27}$   
$\phi$                                 ${3.73}_{-2.59}^{+1.48}$   
$\phi$                                 ${5.05}_{-2.89}^{+0.30}$   
$\theta$                               ${1.77}_{-0.05}^{+0.05}$   
$\psi$                                 ${1.76}_{-0.94}^{+0.75}$   
$\phi_{\mathrm{JL}}$                   ${3.72}_{-2.58}^{+1.21}$   
$\phi_{12}$                            ${3.32}_{-2.33}^{+2.04}$   
$\theta_{\mathrm{JN}}$                 ${0.88}_{-0.45}^{+1.70}$   
$m_f\ [M_{\odot}]$                 ${253.89}_{-11.77}^{+15.41}$   
$\chi_f$                               ${0.70}_{-0.06}^{+0.06}$   
$v_{\rm kick}\ [{\rm km/s}]$     ${708.28}_{-449.99}^{+790.20}$   

                                                                        Prior  
Parameter                                                                      
$m_1\ [M_{\odot}]$                                                        $-$  
$m_2\ [M_{\odot}]$                                                        $-$  
$\mathcal{M}\ [M_{\odot}]$    $\text{UniformInComponentsChirpMass} [12, 400]$  
$q$                           $\text{UniformInComponentsMassRatio} [0.17, 1]$  
$a_1$                                              $\text{Uniform} [0, 0.99]$  
$a_2$                                              $\text{Uniform} [0, 0.99]$  
$\theta_1$                                            $\text{Sine} [0, 3.14]$  
$\theta_2$                                            $\text{Sine} [0, 3.14]$  
$\chi_{\mathrm{eff}}$                                                     $-$  
$\chi_{\mathrm{p}}$                                                       $-$  
$\alpha$                                                                  $-$  
$\delta$                                                                  $-$  
$t_c\ [s]$                    $\text{Uniform} [1242442967.36, 1242442967.56]$  
$d_L\ [Mpc]$                     $\text{UniformSourceFrame} [100.0, 10000.0]$  
$\phi$                                             $\text{Uniform} [0, 6.28]$  
$\phi$                                             $\text{Uniform} [0, 6.28]$  
$\theta$                                              $\text{Sine} [0, 3.14]$  
$\psi$                                             $\text{Uniform} [0, 3.14]$  
$\phi_{\mathrm{JL}}$                               $\text{Uniform} [0, 6.28]$  
$\phi_{12}$                                        $\text{Uniform} [0, 6.28]$  
$\theta_{\mathrm{JN}}$                                $\text{Sine} [0, 3.14]$  
$m_f\ [M_{\odot}]$                                                        $-$  
$\chi_f$                                                                  $-$  
$v_{\rm kick}\ [{\rm km/s}]$                                              $-$



| Parameter                    | Posterior 90% CI                  | Prior                                           |
|:-----------------------------|:----------------------------------|:------------------------------------------------|
| $m_1\ [M_{\odot}]$           | ${150.90}_{-12.31}^{+17.37}$      | $-$                                             |
| $m_2\ [M_{\odot}]$           | ${117.28}_{-22.33}^{+14.55}$      | $-$                                             |
| $\mathcal{M}\ [M_{\odot}]$   | ${114.27}_{-7.85}^{+8.02}$        | $\text{UniformInComponentsChirpMass} [12, 400]$ |
| $q$                          | ${0.78}_{-0.21}^{+0.15}$          | $\text{UniformInComponentsMassRatio} [0.17, 1]$ |
| $a_1$                        | ${0.66}_{-0.43}^{+0.24}$          | $\text{Uniform} [0, 0.99]$                      |
| $a_2$                        | ${0.69}_{-0.42}^{+0.23}$          | $\text{Uniform} [0, 0.99]$                      |
| $\theta_1$                   | ${1.58}_{-0.57}^{+0.56}$          | $\text{Sine} [0, 3.14]$                         |
| $\theta_2$                   | ${1.56}_{-0.61}^{+0.55}$          | $\text{Sine} [0, 3.14]$                         |
| $\chi_{\mathrm{eff}}$        | ${0.00}_{-0.18}^{+0.20}$          | $-$                                             |
| $\chi_{\mathrm{p}}$          | ${0.67}_{-0.25}^{+0.19}$          | $-$                                             |
| $\alpha$                     | ${3.32}_{-0.12}^{+2.60}$          | $-$                                             |
| $\delta$                     | ${0.48}_{-1.38}^{+0.20}$          | $-$                                             |
| $t_c\ [s]$                   | ${1242442967.44}_{-0.04}^{+0.00}$ | $\text{Uniform} [1242442967.36, 1242442967.56]$ |
| $d_L\ [Mpc]$                 | ${4545.93}_{-1668.08}^{+1594.27}$ | $\text{UniformSourceFrame} [100.0, 10000.0]$    |
| $\phi$                       | ${3.73}_{-2.59}^{+1.48}$          | $\text{Uniform} [0, 6.28]$                      |
| $\phi$                       | ${5.05}_{-2.89}^{+0.30}$          | $\text{Uniform} [0, 6.28]$                      |
| $\theta$                     | ${1.77}_{-0.05}^{+0.05}$          | $\text{Sine} [0, 3.14]$                         |
| $\psi$                       | ${1.76}_{-0.94}^{+0.75}$          | $\text{Uniform} [0, 3.14]$                      |
| $\phi_{\mathrm{JL}}$         | ${3.72}_{-2.58}^{+1.21}$          | $\text{Uniform} [0, 6.28]$                      |
| $\phi_{12}$                  | ${3.32}_{-2.33}^{+2.04}$          | $\text{Uniform} [0, 6.28]$                      |
| $\theta_{\mathrm{JN}}$       | ${0.88}_{-0.45}^{+1.70}$          | $\text{Sine} [0, 3.14]$                         |
| $m_f\ [M_{\odot}]$           | ${253.89}_{-11.77}^{+15.41}$      | $-$                                             |
| $\chi_f$                     | ${0.70}_{-0.06}^{+0.06}$          | $-$                                             |
| $v_{\rm kick}\ [{\rm km/s}]$ | ${708.28}_{-449.99}^{+790.20}$    | $-$                                             |


Lets make some plots!

In [5]:
# NRSurrogate corner plots

import os

param_sets = dict(
    mass=["mass_1", "mass_2", "chirp_mass", "mass_ratio"],
    spin=["a_1", "a_2", "tilt_1", "tilt_2"],
    effective_spin=["mass_ratio", "chi_eff", "chi_p"],
    sky_localisation=["luminosity_distance", "ra", "dec"],
    remnant=["final_mass", "final_spin", "final_kick"],
)
for name, params in param_sets.items():
    fname = f"GW190521_030229_{name}_corner.png"
    if not os.path.isfile(fname):
        fig = nrsur_result.plot_corner(params)
        fig.savefig(fname)

    if name == "remnant":
        continue

    # LVK-Comparison plots
    fname = f"GW190521_030229_compare_{name}_corner.png"
    if not os.path.isfile(fname):
        fig = nrsur_result.plot_lvk_comparison_corner(params)
        fig.savefig(fname)


## Corner Plots

### Mass


::::{tab-set}

:::{tab-item} NRSurrogate
:sync: key1

!["GW190521_030229_mass_corner.png"](GW190521_030229_mass_corner.png)
:::

:::{tab-item} LVK-Comparison
:sync: key2

!["GW190521_030229_compare_mass_corner.png"](GW190521_030229_compare_mass_corner.png)
:::

::::



### Spin


::::{tab-set}

:::{tab-item} NRSurrogate
:sync: key1

!["GW190521_030229_spin_corner.png"](GW190521_030229_spin_corner.png)
:::

:::{tab-item} LVK-Comparison
:sync: key2

!["GW190521_030229_compare_spin_corner.png"](GW190521_030229_compare_spin_corner.png)
:::

::::


### Effective Spin


::::{tab-set}

:::{tab-item} NRSurrogate
:sync: key1

!["GW190521_030229_effective_spin_corner.png"](GW190521_030229_effective_spin_corner.png)
:::

:::{tab-item} LVK-Comparison
:sync: key2

!["GW190521_030229_compare_effective_spin_corner.png"](GW190521_030229_compare_effective_spin_corner.png)
:::

::::



### Sky-localisation


::::{tab-set}

:::{tab-item} NRSurrogate
:sync: key1

!["GW190521_030229_sky_localisation_corner.png"](GW190521_030229_sky_localisation_corner.png)
:::

:::{tab-item} LVK-Comparison
:sync: key2

!["GW190521_030229_compare_sky_localisation_corner.png"](GW190521_030229_compare_sky_localisation_corner.png)
:::

::::



### Remnant

!["GW190521_030229_remnant_corner.png"](GW190521_030229_remnant_corner.png)


## Waveform posterior-predictive plot

In [6]:
fname = f"GW190521_030229_waveform.png"
if not os.path.isfile(fname):
    fig = nrsur_result.plot_signal(outdir=".")


![waveform](GW190521_030229_waveform.png)

## Animations  
Here are some animations of the posterior.

|Spin       | 
|-----------|
| <iframe width="630" height="315" src="https://www.youtube.com/embed/g__34yJhHy8?autoplay=1&mute=1&playlist=g__34yJhHy8&loop=1" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>    |   

|Remnant       | 
|-----------|
| <iframe width="630" height="315" src="https://www.youtube.com/embed/ISzaj0pwggo?autoplay=1&mute=1&playlist=ISzaj0pwggo&loop=1" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>    | 


## Analysis configs
Below are the configs used for the analysis of this job.

In [7]:
nrsur_result.print_configs()

Tmax:                               10000
accounting:                         None
accounting_user:                    None
adapt:                              False
allow_tape:                         True
analysis_executable:                None
analysis_executable_parser:         None
autocorr_c:                         5.0
autocorr_tol:                       50.0
bilby_zero_likelihood_mode:         False
burn_in_nact:                       50.0
calibration_model:                  CubicSpline
calibration_prior_boundary:         reflective
catch_waveform_errors:              True
channel_dict:                       {H1:DCS-CALIB_STRAIN_CLEAN_SUB60HZ_C01, L1:DCS-CALIB_STRAIN_CLEAN_SUB60HZ_C01, V1:Hrec_hoft_16384Hz}
check_point_deltaT:                 3600
clean:                              False
coherence_test:                     False
conda_env:                          None
condor_job_priority:                0
conversion_function:                None
create_summary:              

If you used this data, please [cite this work](../citation.md).

## Comments
Leave a comment below or in this [issue](https://github.com/avivajpeyi/nrsur_catlog/issues/new?title=GW190521_030229).
```{raw} html
<script src="https://utteranc.es/client.js"
        repo="avivajpeyi/nrsur_catalog"
        issue-term="GW190521_030229"
        theme="github-light"
        crossorigin="anonymous"
        async>
</script>
```